sys/include/net/ng_netconf.h

```diff
@@ -119,8 +119,12 @@ typedef enum {
      * @note not all transceivers may support this interrupt
      */
     NETCONF_OPT_TX_END_IRQ,
-    NETCONF_OPT_AUTOCCA,            /**< en/disable to check automatically
-                                         before sending the channel is clear. */
+    NETCONF_OPT_CSMA,               /**< en/disable to check automatically
```

Comment:
> But how to ensure that these options don't interfere? I could imagine to implement AUTOCCA for the at86rf2xx by setting minBE = maxBE = 0 and CSMA_RETRIES = 0. But then you shouldn't mess with the CSMA setting.

True, that could be ambiguous. Then maybe we could just introduce the option NETCONF_OPT_CCA for CCA. CSMA could be de/activated by setting the options minBE, maxBE and CSMA_RETRIES. Do you see any option clash there?

> When implementing SW-CSMA, one would try enabling CSMA first, and if that fails, try AUTOCCA I guess.

Yes, thats what I also had in mind.

> What is the bahaviour of kw2xrf? Is it trying resend the packet when channel is not clear? Anyway we need signaling if the packet was actually sent. I have an implementation for that, but didn't PR yet.

> What exactly does the kw2x when its integrated CCA detects that the channel is already occupied?

@OlegHahm @daniel-k The kw2x does not try to resend the packet in that case. It just triggers an interrupt; with a read of the status register one can figure out the ISR-reason then.

In my currently open CSMA_PR (#2467) this is handled by introducing the new event-types NETDEV_EVENT_CCA_CHANNEL_BUSY/IDLE. The driver has to figure out according to the occured interrupts what event it needs to trigger.